### PR TITLE
Reclaim watch and build commands

### PR DESCRIPTION
**Pull request by eggplantzzz**
This reverts the commits responsible for removing the build and watch commands.  There is also some refactoring, syntax updating, test adding and reordering of an index file so the help is in alphabetical order.

### CODE REVIEW

**Comment by eggplantzzz** [comment]
I'm not sure if it is necessary to have the continuous function calling in `continuouslyCheck...`, if it is not there could be some simplification.  I also have some ideas about extracting that function as well as the `print..` methods to make `run` clearer but I don't know how beneficial that is.

Also, do we want to test this?

File: packages/truffle-core/lib/build.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+function CommandBuilder(command) {
+  this.command = command;
+};
+
+CommandBuilder.prototype.build = function(options, callback) {
+  console.log("Running `" + this.command + "`...")
+
+  const args = spawnargs(this.command);
+  const ps = args.shift();
```
Comment (eggplantzzz) [inline packages/truffle-core/lib/build.js]: By the way, what does `ps` stand for?

File: packages/truffle-core/lib/commands/help.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 var command = {
   command: "help",
-  description: "Display a list of all commands or information about a specific command",
+  description: "Display a list of all commands or info about a specific command",
```
Comment (fainashalts) [inline packages/truffle-core/lib/commands/help.js]: why shorten this to "info"? "information" may be more understandable to people for whom English is not their first language since info is slang.

File: packages/truffle-core/lib/build.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
+const async = require("async");
+const mkdirp = require("mkdirp");
+const del = require("del");
+const fs = require("fs");
+const Contracts = require("truffle-workflow-compile");
```
Comment (fainashalts) [inline packages/truffle-core/lib/build.js]: is there a reason the Contracts and BuildError constants are capitalized in the first letter or should they start with a lowercase?

File: packages/truffle-core/lib/build.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+const BuildError = require("./errors/builderror");
+const child_process = require("child_process");
```
Comment (fainashalts) [inline packages/truffle-core/lib/build.js]: should this be childProcess for consistency?

File: packages/truffle-core/lib/build.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+              return callback(new BuildError(err));
+            }
+            return callback(err);
+          });
+        }
+      });
+    });
+  },
+
+  // Deprecated: Specific to default builder.
```
Comment (fainashalts) [inline packages/truffle-core/lib/build.js]: Is this actually deprecated? If so might be a good time to remove.

File: packages/truffle-core/lib/build.js (hunk 3 above)
Comment (gnidan) [reply packages/truffle-core/lib/build.js]: That's just convention. Typically it depends what gets returned... if you're requiring a class or something that looks like a class, UpperCamelCase seems appropriate

File: packages/truffle-core/lib/build.js (hunk 4 above)
Comment (adrianmcli) [reply packages/truffle-core/lib/build.js]: The only usage of it is `spawn`, and in the official [documentation](https://nodejs.org/api/child_process.html), they deconstruct directly to it. So we should probably do the same. For example:

```js
const { spawn } = require('child_process');
```

Comment (gnidan) [reply packages/truffle-core/lib/build.js]: It's annoying that this is what Node calls it. I usually sidestep the issue by requiring specific functions, not all of `child_process`.

File: packages/truffle-core/lib/build.js (hunk 1 above)
Comment (gnidan) [reply packages/truffle-core/lib/build.js]: Process? Search me.

File: packages/truffle-core/lib/errors/builderror.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+var colors = require("colors");
+var TruffleError = require("truffle-error");
+var inherits = require("util").inherits;
```
Comment (fainashalts) [inline packages/truffle-core/lib/errors/builderror.js]: can we change these to const instead of var?

File: packages/truffle/test/scenarios/commands/build.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+const assert = require("assert");
+const CommandRunner = require("../commandrunner");
+const MemoryLogger = require("../memorylogger");
```
Comment (fainashalts) [inline packages/truffle/test/scenarios/commands/build.js]: should these be commandRunner and memoryLogger for snake case? I'm not sure if there's a special reasons for capitalizing the first letter, haven't had my coffee yet!

File: packages/truffle-core/lib/build.js (hunk 3 above)
Comment (fainashalts) [reply packages/truffle-core/lib/build.js]: Ah, that makes sense. Will keep that in mind, thanks!

File: packages/truffle-workflow-compile/index.js [hunk 8]
```diff
@@ -70,7 +70,7 @@ var Contracts = {
     const compilers = (config.compiler)
       ? [config.compiler]
       : Object.keys(config.compilers);
-
+
```
Comment (CruzMolina) [inline packages/truffle-workflow-compile/index.js]: Can you remove the whitespace here?

Comment (adrianmcli) [reply packages/truffle-workflow-compile/index.js]: Should really integrate prettier into the workflow. Someone with more bandwidth should get on that.

File: packages/truffle-core/lib/build.js (hunk 4 above)
Comment (eggplantzzz) [reply packages/truffle-core/lib/build.js]: @adrianmcli Yes, I like deconstructing it.

File: packages/truffle-core/lib/build.js (hunk 5 above)
Comment (eggplantzzz) [reply packages/truffle-core/lib/build.js]: @gnidan Can you comment on this?

File: packages/truffle-core/lib/commands/help.js (hunk 2 above)
Comment (eggplantzzz) [reply packages/truffle-core/lib/commands/help.js]: I changed this to make it fit on one line.  When `truffle help` is used, there previously were too many characters in the string and it wrapped to the next line.  We could try to shorten it some other way...

Comment (gnidan) [reply packages/truffle-core/lib/commands/help.js]: Maybe `List all commands or provide information about a specific command`

File: packages/truffle-core/lib/build.js (hunk 5 above)
Comment (gnidan) [reply packages/truffle-core/lib/build.js]: Is it used? I vote delete

Comment (eggplantzzz) [reply packages/truffle-core/lib/build.js]: It seems like it is unused, I'll get rid of it.

**Comment by eggplantzzz** [comment]
Closes #1294

**Comment by CruzMolina** [comment]
It's been connected, but for clarity, marking #1340 as containing the test steps.
